File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root: return root
        que = [root]
        
        while que:
            curNode = que.pop(0)
            curNode.left , curNode.right = curNode.right , curNode.left
            
            if curNode.left:
                que.append(curNode.left)
            if curNode.right:
                que.append(curNode.right)
        
        return root
        
#         if not root:
#             return root
        
#         left = self.invertTree(root.left)
#         right = self.invertTree(root.right)
               
#         root.left = right
#         root.right = left
        
#         return root

File: 0226-invert-binary-tree/0226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    
//     if(!root) return root;
//     let left = invertTree(root.left);
//     let right = invertTree(root.right);
//     root.left = right;
//     root.right = left;
//     return root
 
    
    let queue = [root];
    if(!queue) return root;
    
    while(queue.length){
        let curVal = queue.pop();
        if(curVal){
               let tmp = curVal.left;
        curVal.left = curVal.right;
        curVal.right = tmp;
        
     
        
        if(curVal.left) queue.push(curVal.left);
        if(curVal.right) queue.push(curVal.right);
        }
        
    }
    return root
}
